### PR TITLE
Try to simlify copyDir

### PR DESCRIPTION
**Pull request by TheOneRing**
This should simplify the copying as we don't need to resolve anything ourselves.
@pasnox @krf

### CODE REVIEW

File: bin/utils.py [hunk 1]
```diff
@@ -517,7 +517,7 @@ def cleanPackageName(basename, packagename):
     return os.path.basename(basename).replace(packagename + "-", "").replace(".py", "")
 
 
-def createSymlink(source, linkName, useAbsolutePath=False):
+def createSymlink(source, linkName, useAbsolutePath=False, target_is_directory=False):
```
Comment (pasnox) [inline bin/utils.py:520]: I would stick to camel cased var name.

File: bin/utils.py [hunk 2]
```diff
@@ -574,37 +574,19 @@ def copyDir(srcdir, destdir, linkOnly=CraftCore.settings.getboolean("General", "
...
-
-                        newLinkName = os.path.join(tmpdir, dirName)
-                        if os.path.exists(newLinkName):
-                            continue
-
-                        # re-create exact same symlink, but this time in the destdir
-                        if createSymlink(os.path.join(tmpdir, linkTo), newLinkName) and copiedFiles != None:
-                            copiedFiles.append(newLinkName)
+        tmpdir = root.replace(srcdir, destdir)
+        for dirName in dirNames:
```
Comment (pasnox) [inline bin/utils.py:578]: Can you re-add check against .svn ? looks like it was problematic copying them.

File: bin/utils.py [hunk 3]
```diff
@@ -574,37 +574,19 @@ def copyDir(srcdir, destdir, linkOnly=CraftCore.settings.getboolean("General", "
...
+            if os.path.islink(os.path.join(root, dirName)):
```
Comment (pasnox) [inline bin/utils.py:579]: I would englobe the dir creation with a try/catch ?

File: bin/utils.py [hunk 4]
```diff
@@ -574,37 +574,19 @@ def copyDir(srcdir, destdir, linkOnly=CraftCore.settings.getboolean("General", "
...
+                # copy the symlinks without resolving them
+                shutil.copy(os.path.join(root, dirName), os.path.join(tmpdir, dirName), follow_symlinks=False)
+            else:
+                os.makedirs(os.path.join(tmpdir, dirName), exist_ok=True)
 
+        for fileName in files:
+            # symlinks to files are included in `files`
+            if copyFile(os.path.join(root, fileName), os.path.join(tmpdir, fileName),
```
Comment (pasnox) [inline bin/utils.py:587]: So copyFile is handling copying a symlink already ?

File: bin/utils.py (hunk 2 above)
Comment (TheOneRing) [reply bin/utils.py:578]: we are able to delete those files for years and we basically have no svn stuff anymore

**Comment by TheOneRing** [comment]
Pushed in 00900cdcb1cde49c0c969453590abb237c3928d6

**Pull request closed without merging**